lib/extras/dec/pnm.cc

```diff
@@ -5,12 +5,16 @@
 
 #include "lib/extras/dec/pnm.h"
 
+#include <fcntl.h>
 #include <stdlib.h>
 #include <string.h>
+#include <sys/mman.h>
```

Comment:
Maybe we can add some abstraction in another file that would handle OS specific stuff.

Reply:
Yeah, that's what I was thinking of doing - the only question is whether we should just error out on non-windows non-unix or if we should emulate it with fread...

Reply:
I think an error message saying that streaming input is not supported would be enough.